azurerm/resource_arm_backup_protected_file_share_test.go

```diff
@@ -0,0 +1,315 @@
+		if err != nil {
+			if utils.ResponseWasNotFound(resp.Response) {
+				return fmt.Errorf("Azure Backup Protected File Share %q (resource group: %q) was not found: %+v", protectedItemName, resourceGroup, err)
+			}
+
+			return fmt.Errorf("Bad: Get on recoveryServicesVaultsClient: %+v", err)
+		}
+
+		return nil
+	}
+}
+
+func testAccAzureRMBackupProtectedFileShare_base(rInt int, location string) string {
+	rstr := strconv.Itoa(rInt)
+	return fmt.Sprintf(` 
+resource "azurerm_resource_group" "test" {
+  name     = "acctestRG-%[1]d"
+  location = "%[2]s"
+}
+
+resource "azurerm_storage_account" "test" {
+  name                     = "acctest%[3]s"
+  location                 = "${azurerm_resource_group.test.location}"
+  resource_group_name      = "${azurerm_resource_group.test.name}"
+  account_tier             = "Standard"
+  account_replication_type = "LRS"
+}
+
+resource "azurerm_storage_share" "test" {
+  name                 = "acctest-%[1]d"
+  storage_account_name = "${azurerm_storage_account.test.name}"
+  metadata             = {}
+  
+  lifecycle {
+	ignore_changes = [metadata] // Ignore changes Azure Backup makes to the metadata
+  }
+}
+
+resource "azurerm_recovery_services_vault" "test" {
+  name                = "acctest-%[1]d"
+  location            = "${azurerm_resource_group.test.location}"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  sku                 = "Standard"
+}
+
+resource "azurerm_backup_policy_file_share" "test1" {
+  name                = "acctest-%[1]d"
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  recovery_vault_name = "${azurerm_recovery_services_vault.test.name}"
+
+  backup {
+    frequency = "Daily"
+    time      = "23:00"
+  }
+
+  retention_daily {
+    count = 10
+  }
+}
+`, rInt, location, rstr[len(rstr)-5:])
+}
+
+func testAccAzureRMBackupProtectedFileShare_basic(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_backup_container_storage_account" "test" {
+  resource_group_name = "${azurerm_resource_group.test.name}"
+  recovery_vault_name = "${azurerm_recovery_services_vault.test.name}"
+  storage_account_id  = "${azurerm_storage_account.test.id}"
+}
+
+resource "azurerm_backup_protected_file_share" "test" {
+  resource_group_name       = "${azurerm_resource_group.test.name}"
+  recovery_vault_name       = "${azurerm_recovery_services_vault.test.name}"
+  source_storage_account_id = "${azurerm_backup_container_storage_account.test.storage_account_id}"
+  source_file_share_name    = "${azurerm_storage_share.test.name}"
+  backup_policy_id          = "${azurerm_backup_policy_file_share.test1.id}"
+}
+`, testAccAzureRMBackupProtectedFileShare_base(rInt, location))
+}
+
+func testAccAzureRMBackupProtectedFileShare_updatePolicy(rInt int, location string) string {
+	return fmt.Sprintf(`
+%s
+
+resource "azurerm_backup_policy_file_share" "test2" {
+  name                = "acctest-%d-secondary"
```

Comment:
Could we put some capitals in here to check the API handles them correctly?
```suggestion
  name                = "acctest-%d-Secondary"
```